yolo/tools/data_loader.py

```diff
@@ -133,6 +142,7 @@ def load_valid_labels(self, label_path: str, seg_data_one_img: list) -> Union[Te
 
     def get_data(self, idx):
         img_path, bboxes = self.data[idx]
+        img_path = self.data_root / Path(img_path)
         img = Image.open(img_path).convert("RGB")
```

Comment:
In the current proposed solution, we are splitting the image_path in `filter_data` the we are completing it back in this function. Such distribution makes the code less predictable and difficult to grasp unless the developers know the code base from the heart. I think we should come up with a better solution.

Reply:
I wonder if it's possible to achieve a similar effect as in yolov8, leaving these paths to be specified by the users. The existing code is overly strict in its requirements for the data format.

Reply:
When open image, it's the same that get the iamge path from image_id.

Reply:
Not sure, if I fully got your point. If you are suggesting that using image_id (image file name minus the extension) would make things easier, I agree.

Reply:
Maybe setting the key to the actual image name with file extension is better?
Saw your comment: https://github.com/WongKinYiu/YOLO/pull/72#discussion_r1717792758
We already have available to us the `dataset_path`, `phase_name` variable. "images" folder name string is common in both coco and yolo datasets. That means we can reconstruct the original path of the image easily if we use the image file name with extension as the key.
`image_path = dataset_path / "Images" / phase_name / image_file_name`

Extions will help resolve .jpg/.png ambiguity.